codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpData.java

```diff
@@ -122,14 +122,16 @@
 
     /**
      * Returns the contents of the file item as an array of bytes.
+     * One should take care of memory allocation if it is a huge content.
      *
      * @return the contents of the file item as an array of bytes.
      * @throws IOException
      */
     byte[] get() throws IOException;
 
     /**
-     * Returns the content of the file item as a ByteBuf
+     * Returns the content of the file item as a ByteBuf.
+     * One should take care of memory allocation if it is a huge content.
```

Comment:
The English for these two comments reads weird. How about:

> Note: this method will allocate a lot of memory, if the data is currently stored on the file system.

Reply:
It's the reverse. I would then propose:

     Note: this method will allocate a lot of memory, if the data is not currently stored on the file system.

Reply:
In `AbstractMemoryHttpData`, the `getByteBuf` method just returns the contained `ByteBuf` instance, so no memory is allocated just from calling that method. Though for the `get` method, it does allocate a new byte array.

For `AbstractDiskHttpData`, both of those methods allocate a byte array.

So the only case that doesn't allocate a lot is `getByteBuf` on `AbstractMemoryHttpData`. So maybe the warning should be unconditional.

Reply:
Yes right ! I just write the exact opposite of my own logic, and the reason I wrote this warning.

Reply:
Done